parsers/PA.py

```diff
@@ -313,44 +245,25 @@ def fetch_exchange(
             sorted_zone_keys,
         )
 
-    df = pd.read_html(response.text)[0]
+    response_payload = response.json()
+    nodes = response_payload["nodes"]
+    nodes_flattened = [node["from"] for node in nodes] + [node["to"] for node in nodes]
 
-    # A positive value on website indicates a flow from country specified to PA.
-    net_flow_cr = round(
-        float(df[4][1])
-        + float(df[4][3])
-        + float(df[4][5])
-        + float(df[1][8])
-        + float(df[1][10]),
-        2,
-    )
-    net_flow_gt = round(
-        float(df[4][23]) + float(df[4][26]) + float(df[4][28]) + float(df[1][31]), 2
-    )
-    net_flow_hn = round(
-        float(df[1][13])
-        + float(df[1][15])
-        + float(df[1][18])
-        + float(df[1][20])
-        + float(df[1][23]),
-        2,
-    )
-    net_flow_ni = round(
-        float(df[4][8]) + float(df[4][10]) + float(df[4][13]) + float(df[4][15]), 2
-    )
-
-    # invert sign to account for direction in alphabetical order
-    net_flow_sv = -1 * round(
-        float(df[4][18]) + float(df[4][20]) + float(df[1][26]) + float(df[1][28]), 2
-    )
+    net_flows = {}
+    for country in COUNTRY_TO_EXCHANGE_NODES:
+        net_flow_country = sum(
+            float(n["mw"])
+            for n in nodes_flattened
+            if n["name"] in COUNTRY_TO_EXCHANGE_NODES[country]
+        )
 
-    net_flows = {
-        "CR->PA": net_flow_cr,  # Costa Rica to Panama
-        "GT->PA": net_flow_gt,  # Guatemala to Panama
-        "HN->PA": net_flow_hn,  # Honduras to Panama
-        "NI->PA": net_flow_ni,  # Nicaragua to Panama
-        "PA->SV": net_flow_sv,  # Panama to El Salvador
-    }
+        # insert net flow taking into account alphabetical sorting of zone keys
+        sorted_exchange_key = ZoneKey("->".join(sorted([country, "PA"])))
+        net_flows[sorted_exchange_key] = (
+            net_flow_country
+            if sorted_exchange_key.endswith("PA")
+            else -net_flow_country
+        )
```

Comment:
issue(maybe): not an issue for this PR (also because I don't think that there is any exchange using this parser) but I don't think that this function is / was calculating the correct things (unless I am understanding things wrong 😅 )

Looking at https://sitr.cnd.com.pa/m/pub/int.html, the flows are between neighbouring countries, not all to `PA`. Also if we tried `fetch_exchange()` on any other combination than "CR->PA" it indeed fails saying that those are not valid exchanges.

Should I just remove this function? If not also happy to try fixing it in another PR

Reply:
Since this is unused it would be fine to remove it but if you want to fix it that would also be great 🙂

Totally up to you though IMO.